src/Octoshift/GithubApi.cs

```diff
@@ -685,6 +686,95 @@ public virtual async Task<MannequinReclaimResult> ReclaimMannequin(string orgId,
+            return await _client.GetAllAsync(url)
+                .Select(codescan => BuildCodeScanningAnalysis(codescan))
+                .ToListAsync();
+        }
+        
+        public virtual async Task UpdateCodeScanningAlert(string org, string repo, int alertNumber, string state, string dismissedReason = null, string dismissedComment = null)
+        {
+            if (!CodeScanningAlerts.IsOpenOrDismissed(state))
+            {
+                throw new ArgumentException($"Invalid value for {nameof(state)}");
+            }
+            
+            if (CodeScanningAlerts.IsDismissed(state) && !CodeScanningAlerts.IsValidDismissedReason(dismissedReason))
+            {
+                throw new ArgumentException($"Invalid value for {nameof(dismissedReason)}");
+            }
+            
+            var url = $"{_apiUrl}/repos/{org}/{repo}/code-scanning/alerts/{alertNumber}";
+
+            object payload;
+            if (state == "open")
+            {
+                payload = new { state };
+            }
+            else
+            {
+                payload = new
+                {
+                    state,
+                    dismissed_reason = dismissedReason,
+                    dismissed_comment = dismissedComment
+                };
+            }
+            await _client.PatchAsync(url, payload);
+        }
+
+        public virtual async Task<IEnumerable<SecretScanningAlert>> GetSecretScanningAlertsForRepository(string org, string repo)
+        {
+            var url = $"{_apiUrl}/repos/{org}/{repo}/secret-scanning/alerts?per_page=100";
+            return await _client.GetAllAsync(url)
+                .Select(secretAlert => BuildSecretScanningAlert(secretAlert))
+                .ToListAsync();
+        }
+        
+        public virtual async Task<IEnumerable<SecretScanningAlertLocation>> GetSecretScanningAlertsLocations(string org, string repo, int alertNumber)
+        {
+            var url = $"{_apiUrl}/repos/{org}/{repo}/secret-scanning/alerts/{alertNumber}/locations?per_page=100";
+            return await _client.GetAllAsync(url)
+                .Select(alertLocation => BuildSecretScanningAlertLocation(alertLocation))
+                .ToListAsync();
+        }
+        
+        public virtual async Task UpdateSecretScanningAlert(string org, string repo, int alertNumber, string state, string resolution = null)
+        {
+            if (!SecretScanningAlerts.IsOpenOrResolved(state))
+            {
+                throw new ArgumentException($"Invalid value for {nameof(state)}");
+            }
+            
+            if (SecretScanningAlerts.IsResolved(state) && !SecretScanningAlerts.IsValidDismissedReason(resolution))
+            {
+                throw new ArgumentException($"Invalid value for {nameof(resolution)}");
+            }
+            
+            var url = $"{_apiUrl}/repos/{org}/{repo}/secret-scanning/alerts/{alertNumber}";
+
+            dynamic payload;
```

Comment:
`payload` should be an `object`:
```suggestion
            object payload;
```